src/Util/ClassSourceManipulator.php

```diff
@@ -197,8 +233,36 @@ public function addSetter(string $propertyName, $type, bool $isNullable, array $
         $this->addMethod($builder->getNode());
     }
 
-    public function addMethodBuilder(Builder\Method $methodBuilder)
+    public function addConstructor(array $params = [], string $methodBody = null)
     {
+        if ($this->getConstructorNode()) {
+            throw new \LogicException('Constructor already exists.', 1);
+        }
+
+        $methodBuilder = $this->createMethodBuilder('__construct', null, false, []);
+
+        if (!empty($params)) {
+            $this->addMethodParams($methodBuilder, $params);
+        }
+
+        if ($methodBody) {
```

Comment:
Can we remove this if? If, for some reason, no methodBody were passed, do we get an error?

Reply:
Yes, done. Yes, an error will be caught